app/services/foreman_ansible/playbooks_importer.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  # Imports playbooks from smart proxy
+  class PlaybooksImporter
+    include ::ForemanAnsible::ProxyAPI
+    delegate :playbooks, :playbooks_names, :to => :proxy_api
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def import_playbooks(playbooks_names)
+      filter_playbooks = playbooks_names.empty? ? playbooks : PlaybooksImporterHelper.find_playbooks(playbooks, playbooks_names)
+      import_playbooks_content filter_playbooks
+    end
+
+    def import_playbooks_content(playbooks)
+      params = { changed: { update: [], create: [] } }
+      playbooks.each do |playbook|
+        new_playbook = parse_playbook playbook
+        job_template = JobTemplate.find_by(name: playbook['name'])
+        old_var_block = PlaybooksImporterHelper.check_vars_block new_playbook[:old_vars_block]
+        if job_template.present?
+          params[:changed][:update].push(
+            {
+              job_template: job_template,
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        else
+          params[:changed][:create].push(
+            {
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        end
+      end
+      params
+    end
+
+    def parse_playbook(playbook)
+      playbook_name =  playbook['name']
+      metadata = add_metadata playbook['name']
+      new_playbook_content = convert_to_job_template_input playbook['playbooks_content']
+      {
+        name: playbook_name,
+        playbook_content: metadata + new_playbook_content[:playbook_content],
+        template_inputs: new_playbook_content[:template_inputs],
+        old_vars_block: new_playbook_content[:old_vars_block]
+      }
+    end
+
+    def add_metadata(playbook_name)
+      "<%#
+      name: #{playbook_name}
+      snippet: false
+      job_category: Ansible Playbook - Imported
+      provider_type: Ansible
+      kind: job_template
+      model: JobTemplate
+      %>\n"
```

Comment:
Since this is one long string spanning across multiple lines, the whitespace at the beginning of the line is also included and the metadata in template seem a little bit misaligned:

![metadata](https://user-images.githubusercontent.com/2664090/142375445-8d6eaa21-8f0e-4f60-925d-a8d14c2e128d.png)

So we might think of how to make sure we get the formatting we want, perhaps [heredoc](https://www.rubyguides.com/2018/11/ruby-heredoc/) might help.